test/testResourceQuoteStyle.js

```diff
@@ -0,0 +1,728 @@
+
+        test.done();
+    },
+
+    testResourceQuoteStyleDontMatchApostrophes: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "quote.test",
+                sourceLocale: "en-US",
+                source: "This string doesn't contain quotes in it.",
+                targetLocale: "de-DE",
+                target: "Diese Zeichenfolge enthält nicht Anführungszeichen.",
+                pathName: "a/b/c.xliff"
+            }),
+            file: "x/y"
+        });
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceQuoteStyleDontMatchMultipleApostrophes: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "quote.test",
+                sourceLocale: "en-US",
+                source: "This string doesn't contain quotes in it. The user's keyboard is working",
+                targetLocale: "de-DE",
+                target: "Diese Zeichenfolge enthält nicht Anführungszeichen. Der Tastenbord des Users funktioniert.",
+                pathName: "a/b/c.xliff"
+            }),
+            file: "x/y"
+        });
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceQuoteStyleApostropheInTarget: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "fr-FR",
+            resource: new ResourceString({
+                key: "quote.test",
+                sourceLocale: "en-US",
+                source: 'This string contains "quotes" in it.',
+                targetLocale: "fr-FR",
+                target: "L'expression contient de «guillemets». C'est tres bizarre !",
+                pathName: "a/b/c.xliff"
+            }),
+            file: "x/y"
+        });
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceQuoteStyleApostropheInTargetSpace: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "fr-FR",
+            resource: new ResourceString({
+                key: "quote.test",
+                sourceLocale: "en-US",
+                source: 'This string contains "quotes" in it.',
+                targetLocale: "fr-FR",
+                target: "L'expression contient de « guillemets ». C'est tres bizarre !",
+                pathName: "a/b/c.xliff"
+            }),
+            file: "x/y"
+        });
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceQuoteStyleApostropheInTargetSpace: function(test) {
```

Comment:
duplicate test?

Reply:
Similar to the one above, but not a duplicate. This one has the apostrophe in it "L'expression" which confused this rule before the fix. Condition is: source has a quote in it, causing it to check the target. The target also has a quote in it that matches, so no problem there. But, target also has the single quote used as an apostrophe, which got flagged as an incorrect quote character. It was not properly checking that the quotes in the target were at the beginning/end of words or beginning/end of the whole string. Out of all of the tests added, this one is most similar to the erroneously flagged error I saw in the Lilt translations.

Reply:
Oooh I see. Then it should have a different name than the previous one

Reply:
Name updated. The actual reason is actually different the first has regular spaces between the guillemets and the word "guillemets", and the second has a non-breaking space there intstead.